### PR TITLE
Add support for Asp.net triggers

### PR DESCRIPTION
**Pull request by wiktork**
Depends on https://github.com/dotnet/diagnostics/pull/2592

### CODE REVIEW

File: src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs [hunk 1]
```diff
@@ -14,10 +19,45 @@ namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers
...
+                StatusCodes = options.StatusCodes.Select(ParseRange).ToArray(),
+                SlidingWindowDuration = options.SlidingWindowDuration ?? TimeSpan.Parse(AspNetResponseStatusOptionsDefaults.SlidingWindowDuration, CultureInfo.InvariantCulture),
+            };
+
+            var aspnetTriggerSourceConfiguration = new AspNetTriggerSourceConfiguration();
+
+            return _eventPipeTriggerFactory.Create(endpointInfo, aspnetTriggerSourceConfiguration, _traceEventTriggerFactory, settings, callback);
+        }
+
+        private static StatusCodeRange ParseRange(string range)
```
Comment (kelltrick) [inline src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs:51]: I think this is a good start but I imagine we may want to expand this syntax to support something like: `200,202-204`

**Review by kelltrick — APPROVED** [review]
👍

File: src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs (hunk 1 above)
Comment (jander-msft) [reply src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs:51]: Each item in the StatusCodes property above represents either a single code or a code range. An individual item should not have a combination of them. So this is permissable:

```json
[ "200", "202-204" ]
```

but not:

```json
[ "200,202-204" ]
```

**Review by jander-msft — APPROVED** [review]

File: src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs (hunk 1 above)
Comment (wiktork) [reply src/Tools/dotnet-monitor/CollectionRules/Triggers/AspNetResponseStatusTriggerFactory.cs:51]: I will leave it as is for now, but if there's strong favor for having the payload be one string I can change it.

**Pull request merged**